src/test/java/org/mitre/synthea/world/concepts/TerminologyTest.java

```diff
@@ -0,0 +1,288 @@
+        assertTrue(codeMap.get("foo").stream().anyMatch(code -> code.code.equals("7")));
+        assertFalse(codeMap.get("foo").stream().anyMatch(code -> code.code.equals("3")));
+
+        assertTrue(codeMap.get("foobar").stream().anyMatch(code -> code.code.equals("2")));
+        assertFalse(codeMap.get("foobar").stream().anyMatch(code -> code.code.equals("5")));
+    }
+
+    @Test
+    public void getClient() {
+        OkHttpClient hello = new OkHttpClient.Builder().build();
+        OkHttpClient goodbye = Terminology.getClient();
+        assertEquals(hello.getClass(), goodbye.getClass());
+    }
+
+    @Test
+    public void getClient1() {
+        OkHttpClient client = Terminology.getClient("www.hello.com", 8080);
+        assertTrue(client.proxy().address().toString().contains("www.hello.com"));
+        assertTrue(client.proxy().address().toString().contains("8080"));
+    }
+
+    @Test
+    public void getAuthToken() {
+        OkHttpClient hello;
+        String PORT = Config.get("terminology.PORT");
+        String HOSTNAME = Config.get("terminology.PROXY_HOSTNAME");
+        if(HOSTNAME!=null & PORT!=null){
+            hello = new OkHttpClient.Builder()
+                    .proxy(new Proxy(Proxy.Type.HTTP,
+                            new InetSocketAddress(HOSTNAME, Integer.parseInt(PORT))))
+                    .build();
+        }else{
+            hello = new OkHttpClient.Builder()
+                    .build();
+        }
+        if(Config.get("VSAC_USER")!=null & Config.get("VSAC_PASS")!=null){
+
+            String authToken = Terminology.getAuthToken(hello);
+            assertTrue(authToken.length() > 10);
+            assertNotNull(authToken);
+        }
+
+
+    }
+
+    @Test
+    public void getServiceTicket() {
+        if(session.getAuthToken()!=null){
+            String token = Terminology.getServiceTicket(session.getAuthToken(), session.getSessionClient());
+            assertTrue(token.length() > 10);
+            assertNotNull(token);
+        }
+    }
+
+
+    // These functions aren't used yet
+
+//    @Test
+//    public void makeGenericPost() {
+//    }
+//
+//    @Test
+//    public void makeGenericGet() {
+//    }
+
+    @Test
+    public void makeRequest() {
+        String url = "http://echo.jsontest.com/title/ipsum/content/blah";
+        Request request = Terminology.makeRequest(url);
+        assertEquals(request.url().toString(), url);
+        try {
+            Response response = session.getSessionClient().newCall(request).execute();
+
+            assertTrue(Objects.requireNonNull(response.body()).string().contains("ipsum"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+    }
+
+    @Test
+    public void makeRequest1() {
+        FormBody formBody = new FormBody.Builder().add("json", "{1:2}").build();
+        Request request = Terminology.makeRequest("http://validate.jsontest.com/", formBody);
```

Comment:
I don't want real HTTP requests in the unit tests. Use a fake address with Mocks. For example, https://github.com/square/okhttp/tree/master/mockwebserver